src/dsp_tools/utils/excel2json/utils.py

```diff
@@ -284,3 +284,26 @@ def col_must_or_not_empty_based_on_other_col(
         return pd.Series(combined_array)
     else:
         return None
+
+
+def add_optional_columns(df: pd.DataFrame, optional_col_set: set[str]) -> pd.DataFrame:
+    """
+    This function takes a df and a set of columns which may not be in the df, but whose absence could cause errors in
+    the code following.
+    As a quick fix, they are added to the df.
```

Comment:
"Quick fix" is a term used in work organization, but it seems inappropriate in a function docstring. Because the docstring should describe what the function does, no matter if it is a quick fix or not.